src/main/java/uk/gov/hmcts/reform/sscs/service/hearings/CreateHearingCaseUpdater.java

```diff
@@ -0,0 +1,151 @@
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.service.SscsCcdConvertService;
+import uk.gov.hmcts.reform.sscs.ccd.service.UpdateCcdCaseService;
+import uk.gov.hmcts.reform.sscs.exception.GetHearingException;
+import uk.gov.hmcts.reform.sscs.exception.ListingException;
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.helper.mapping.OverridesMapping;
+import uk.gov.hmcts.reform.sscs.helper.service.HearingsServiceHelper;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.model.HearingEvent;
+import uk.gov.hmcts.reform.sscs.model.HearingWrapper;
+import uk.gov.hmcts.reform.sscs.model.hearings.HearingRequest;
+import uk.gov.hmcts.reform.sscs.model.multi.hearing.CaseHearing;
+import uk.gov.hmcts.reform.sscs.model.multi.hearing.HearingsGetResponse;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingGetResponse;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingRequestPayload;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HmcUpdateResponse;
+import uk.gov.hmcts.reform.sscs.service.HmcHearingApiService;
+import uk.gov.hmcts.reform.sscs.service.HmcHearingsApiService;
+import uk.gov.hmcts.reform.sscs.service.holder.ReferenceDataServiceHolder;
+
+import static java.util.Objects.isNull;
+import static uk.gov.hmcts.reform.sscs.helper.mapping.HearingsMapping.buildHearingPayload;
+
+@Slf4j
+@Component
+public class CreateHearingCaseUpdater extends HearingSaveActionBase {
+
+    private final HmcHearingApiService hmcHearingApiService;
+
+    private final HmcHearingsApiService hmcHearingsApiService;
+
+    private final ReferenceDataServiceHolder refData;
+
+    private final IdamService idamService;
+
+    private static final Long HEARING_VERSION_NUMBER = 1L;
+
+    @Autowired
+    public CreateHearingCaseUpdater(CcdClient ccdClient, SscsCcdConvertService sscsCcdConvertService, HmcHearingApiService hmcHearingApiService,
+                                    HmcHearingsApiService hmcHearingsApiService, ReferenceDataServiceHolder refData,
+                                    IdamService idamService) {
+        super(ccdClient, sscsCcdConvertService, refData);
+        this.hmcHearingApiService = hmcHearingApiService;
+        this.hmcHearingsApiService = hmcHearingsApiService;
+        this.refData = refData;
+        this.idamService = idamService;
+    }
+
+    public void createHearingAndUpdateCase(HearingRequest hearingRequest) throws ListingException, UpdateCaseException {
+        HearingEvent event = HearingsServiceHelper.getHearingEvent(hearingRequest.getHearingState());
+        try {
+            updateCase(
+                Long.valueOf(hearingRequest.getCcdCaseId()),
+                event.getEventType().getCcdType(),
+                idamService.getIdamTokens(),
+                hearingRequest
+            );
+
```

Comment:
Should this log be added here:
if (nonNull(details)) {
            log.info("Case update details CCD state {}  event id: {} event token: {} callbackresponsestatus: {} caseid {}",
                details.getState(),
                details.getEventId(),
                details.getEventToken(),
                details.getCallbackResponseStatus(),
                details.getCaseTypeId()
            );
        }